meta/collection.go

```diff
@@ -0,0 +1,410 @@
+									End:   recorder.End,
+								},
+							},
+						},
+						Stream:                 stream,
+						Gains:                  gains,
+						SensorCalibrations:     sensors,
+						DataloggerCalibrations: dataloggers,
+						Channel:                channel,
+						Component:              component,
+						Span:                   span,
+					})
+				}
+			}
+		}
+	}
+
+	for _, connection := range s.Connections() {
+		if connection.Station != site.Station {
+			continue
+		}
+		if connection.Location != site.Location {
+			continue
+		}
+
+		for _, sensor := range s.InstalledSensors() {
+			if sensor.Station != site.Station {
+				continue
+			}
+			if sensor.Location != site.Location {
+				continue
+			}
+
+			for _, component := range s.Components() {
+				if sensor.Make != component.Make {
+					continue
+				}
+				if sensor.Model != component.Model {
+					continue
+				}
+
+				for _, datalogger := range s.DeployedDataloggers() {
+					if datalogger.Place != connection.Place {
+						continue
+					}
+					if datalogger.Role != connection.Role {
+						continue
+					}
+
+					span, ok := connection.Span.Extent(sensor.Span, datalogger.Span)
+					if !ok {
+						continue
+					}
+
+					for _, stream := range s.Streams() {
+						if stream.Station != site.Station {
+							continue
+						}
+						if stream.Location != site.Location {
+							continue
+						}
+
+						span, ok := span.Extent(stream.Span)
+						if !ok {
+							continue
+						}
+
+						for _, channel := range s.Channels() {
+							if datalogger.Make != channel.Make {
+								continue
+							}
+							if datalogger.Model != channel.Model {
+								continue
+							}
+							if component.Number+connection.Number < channel.Number {
+								continue
+							}
+
+							if stream.SamplingRate != channel.SamplingRate {
+								continue
+							}
+
+							var gains []Gain
+							for _, g := range s.Gains() {
+								if g.Station != stream.Station {
+									continue
+								}
+								if g.Location != stream.Location {
+									continue
+								}
+								if g.Subsource != component.Subsource {
+									continue
+								}
+								if !span.Overlaps(g.Span) {
+									continue
+								}
+								gains = append(gains, g)
+							}
+							sort.Slice(gains, func(i, j int) bool {
+								return gains[i].Span.Start.Before(gains[j].Span.Start)
+							})
+
+							var sensors []Calibration
+							for _, c := range s.Calibrations() {
+								if c.Make != sensor.Make {
+									continue
+								}
+								if c.Model != sensor.Model {
+									continue
+								}
+								if c.Serial != sensor.Serial {
+									continue
+								}
+								if c.Number != component.Number {
+									continue
+								}
+								if !span.Overlaps(c.Span) {
+									continue
+								}
+								sensors = append(sensors, c)
+							}
+							sort.Slice(sensors, func(i, j int) bool {
+								return sensors[i].Span.Start.Before(sensors[j].Span.Start)
+							})
```

Comment:
similar to the above, check if it is possible to calculate `gains` and `sensors` out side the channels loop